src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt

```diff
@@ -0,0 +1,72 @@
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumFingerprint
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumParentFingerprint
+import app.revanced.patches.youtube.layout.startupshortsreset.annotations.StartupShortsResetCompatibility
+import app.revanced.patches.youtube.layout.startupshortsreset.fingerprints.ActionOpenShortsFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("disable-startup-shorts-player")
+@Description("Disable playing YouTube Shorts when launching YouTube.")
+@StartupShortsResetCompatibility
+@Version("0.0.1")
+class DisableShortsOnStartupPatch : BytecodePatch(
+    listOf(
+        ActionOpenShortsFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_startup_shorts_player_enabled",
+                StringResource("revanced_startup_shorts_player_title", "Disable shorts player at app startup"),
+                false,
+                StringResource("revanced_startup_shorts_player_summary_on", "Shorts player is disabled at app startup"),
+                StringResource("revanced_startup_shorts_player_summary_off", "Shorts player is enabled at app startup")
+            )
+        )
+
+        val actionOpenShortsResult = ActionOpenShortsFingerprint.result
+        val actionOpenShortsMethod = actionOpenShortsResult!!.mutableMethod
+        val actionOpenShortsIstructions = actionOpenShortsMethod.implementation!!.instructions
+
+        val moveResultIndex = actionOpenShortsResult.scanResult.patternScanResult!!.endIndex + 21
+        val iPutBooleanIndex = moveResultIndex + 2
+
+        val moveResultRegister = (actionOpenShortsIstructions[moveResultIndex] as OneRegisterInstruction).registerA
+        val iPutBooleanRegister = (actionOpenShortsIstructions[iPutBooleanIndex] as TwoRegisterInstruction).registerA
+
+        actionOpenShortsMethod.addInstructions(
+            moveResultIndex + 1, """
+            invoke-static { }, Lapp/revanced/integrations/patches/DisableStartupShortsPlayerPatch;->disableStartupShortsPlayer()Z
+            move-result v1
```

Comment:
Instead of allocating the complexity to a patch, simply proxy the original register value and return it from the proxy method.